### PR TITLE
Add and Delete items -- in-process -- DO NOT MERGE

### PR DESCRIPTION
**Pull request by aaronkoz74**
@bjblock -- stuck on this.  Everything I seem to do either allows any user's items to be manipulated (add or delete) or all access to adding and deleting items is removed (for the signed-in user too).  Tried various `before_action` variations in the `ItemsController`.  Any direction you can provide would be appreciated.

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -16,4 +17,25 @@ def create
...
+  end
+
+  private
+
+  def item_params
+    params.require(:item).permit(:description)
+  end
+
+  def only_permit_current_user
+    unless user_signed_in? && (@user == current_user)
```
Comment (bjblock) [inline app/controllers/items_controller.rb]: in order to make this work, you need to first figure out who `@user` is (e.g. `@user = User.find(params[:user_id])`).  Another option would be to ignore the params[:user_id] entirely when creating new items and always set the item's user as the current user.

Comment (bjblock) [inline app/controllers/items_controller.rb]: we can go over this tomorrow.

Comment (aaronkoz74) [inline app/controllers/items_controller.rb]: Got it to work.

For some reason I can't get the ajax function to work properly ... removed
the `redirect_to` portion in the `destroy` method and my page isn't
refreshing without a manual command.

Talk to you in a while.

Thanks

Comment (bjblock) [inline app/controllers/items_controller.rb]: 3 steps for an AJAX call:
1. the view needs to know to make a remote AJAX call when a link is clicked
   or a form is submitted, which means it needs `remote: true`
2. the action in the controller needs to know that it should be able to
   respond to an AJAX request, which means it needs some sort of `respond_to`
   block with an option for `js`
3. you need a js view template with what js that will be executed once the
   request is successful, which means you need a file titled `action.js.erb`
   where "action" is the name of the controller action (e.g. "create") and, in
   that file, you need your js to change the view in some way (e.g. by
   appending to something, or prepending to something, or removing something,
   or adding something, etc...)

**Pull request merged**